### PR TITLE
Connectivity check bug fix.

### PR DESCRIPTION
**Pull request by smaheshwaram**
### Description:

This PR is a bug fix for existing connectivity check. The code checks for synthetic boundary nodes and if the nodes are not one of them it creates flags. Other validations are left as is.

### Potential Impact:

Changes made to Atlas-checks.

### Unit Test Approach:

Added one unit test for validating flags with added synthetic node.

### Test Results:

Using some Atlas files, verified that the starting node and near by nodes are checking the synthetic boundary condition which is added and filter on it. Also, will take analyst help to analyze the flags correctly.

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/checks/validation/points/ConnectivityCheck.java [hunk 1]
```diff
@@ -103,9 +104,11 @@ public boolean validCheckForObject(final AtlasObject object)
                         && this.getLayerMap(nearbyNode).keySet().stream()
                                 .anyMatch(nodeLayerMap::containsKey)))
         {
-            // Flag nearby nodes if they are not are start node, are not a barrier, have a valid
+            // Flag nearby nodes if they are not synthetic boundary node, are start node, are not a
```
Comment (danielduhh) [inline src/main/java/org/openstreetmap/atlas/checks/validation/points/ConnectivityCheck.java]: typo: are not synthetic boundary or start nodes**

File: src/main/java/org/openstreetmap/atlas/checks/validation/points/ConnectivityCheck.java [hunk 2]
```diff
@@ -68,8 +69,8 @@ public ConnectivityCheck(final Configuration configuration)
     @Override
     public boolean validCheckForObject(final AtlasObject object)
     {
-        return object instanceof Node && !this.isFlagged(object.getOsmIdentifier())
-                && !BarrierTag.isBarrier(object)
+        return object instanceof Node && !SyntheticBoundaryNodeTag.isSyntheticBoundaryNode(object)
```
Comment (danielduhh) [inline src/main/java/org/openstreetmap/atlas/checks/validation/points/ConnectivityCheck.java]: Could you add a test case for this ?

Comment (smaheshwaram) [reply src/main/java/org/openstreetmap/atlas/checks/validation/points/ConnectivityCheck.java]: Done
